internal/client/config.go

```diff
@@ -260,10 +273,18 @@ func WriteConfig(filename string, server string, tlsServerName string, ca *crypt
 		ClientKeyData:         clientKeyPEM,
 	}
 
-	contents, err := yaml.Marshal(config)
+	return config.Persist(filename)
+}
+
+func (c *Config) Persist(filename string) error {
+	contents, err := yaml.Marshal(c)
 	if err != nil {
 		return fmt.Errorf("encoding config: %v", err)
 	}
+	directory := filename[:strings.LastIndex(filename, "/")]
+	if err := os.MkdirAll(directory, 0700); err != nil {
+		return fmt.Errorf("writing config: %v", err)
```

Comment:
minor: please wrap errors

```
		return fmt.Errorf("writing config: %w", err)
```

Reply:
fixed, i've changed also other errors in this file